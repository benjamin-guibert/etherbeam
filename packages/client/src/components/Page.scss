@use '../styles/resources/sizes';

.my-page {
  margin: sizes.$space sizes.$space-l;
  position: relative;

  &-titleloader {
    position: absolute;
    top: sizes.$space;
    left: sizes.$space-s + sizes.$space;
  }

  &-subheader {
    display: flex;
  }
}
