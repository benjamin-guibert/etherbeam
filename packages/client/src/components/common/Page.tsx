import React, { ReactElement, ReactNode } from 'react'
import Container from 'react-bootstrap/Container'

interface PageProps {
  title: string
  children: ReactNode
}

const Page = ({ title, children }: PageProps): ReactElement => {
  return (
    <Container>
      <h1>{title}</h1>
      {children}
    </Container>
  )
}

export default Page
