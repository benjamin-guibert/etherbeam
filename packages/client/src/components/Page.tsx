import React, { ReactElement, ReactNode } from 'react'
import Loader from './Loader'
import './Page.scss'

type WidthLimitProp = 'xs' | 's' | 'm' | 'l'

interface PageProps {
  title: string | ReactNode
  widthLimit?: WidthLimitProp
  titleLoading?: boolean
  children?: ReactNode
}

const Page = ({ title, children, widthLimit = 's', titleLoading }: PageProps): ReactElement => {
  return (
    <main className={`my-page ${widthLimit ? `my-mw-${widthLimit}` : ''}`}>
      <h1 className="my-page-header">
        {titleLoading ? (
          <div className="my-page-titleloader">
            <Loader type="dots" />
          </div>
        ) : (
          title
        )}
      </h1>
      {children}
    </main>
  )
}

export default Page
