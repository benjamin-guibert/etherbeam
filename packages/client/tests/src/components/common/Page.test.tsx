import * as React from 'react'
import { mount, ReactWrapper } from 'enzyme'
import Page from 'components/common/Page'

describe('<Page />', () => {
  let component: ReactWrapper

  beforeAll(() => {
    component = mount(<Page title="Title">Content</Page>)
  })

  it('should render', () => {
    const header = component.find('h1')
    expect(header.prop('children')).toEqual('Title')
  })

  it('should match snapshot', () => expect(component.render()).toMatchSnapshot())
})
