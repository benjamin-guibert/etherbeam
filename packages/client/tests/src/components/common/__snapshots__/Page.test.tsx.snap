// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> should match snapshot 1`] = `
<div
  class="container"
>
  <h1>
    Title
  </h1>
  Content
</div>
`;
